### PR TITLE
test: add ANTLR parser crash regression test

### PR DESCRIPTION
**Pull request by kyessenov**
cc @mandarjog

### CODE REVIEW

File: test/envoye2e/stats_plugin/stats_test.go [hunk 1]
```diff
@@ -365,3 +365,37 @@ func TestAttributeGen(t *testing.T) {
 		})
 	}
 }
+
+func TestStatsParserRegression(t *testing.T) {
+	// This is a regression test for https://github.com/envoyproxy/envoy-wasm/issues/497
+	params := driver.NewTestParams(t, map[string]string{
+		"StatsFilterCode":         "inline_string: \"envoy.wasm.stats\"",
+		"WasmRuntime":             "envoy.wasm.runtime.null",
```
Comment (PiotrSikora) [inline test/envoye2e/stats_plugin/stats_test.go:373]: This happens even with NullVm?

Comment (kyessenov) [reply test/envoye2e/stats_plugin/stats_test.go:373]: Yes. It's an issue with the host, not the module.

**Comment by kyessenov** [comment]
I'm waiting for it to crash on CI. Then I'll pull an update for Envoy that fixes it.

File: test/envoye2e/stats_plugin/stats_test.go [hunk 2]
```diff
@@ -365,3 +365,37 @@ func TestAttributeGen(t *testing.T) {
...
+		"StatsConfig":             driver.LoadTestData("testdata/bootstrap/stats.yaml.tmpl"),
+		"ClientHTTPFilters":       driver.LoadTestData("testdata/filters/stats_outbound.yaml.tmpl"),
+		"StatsFilterClientConfig": "{}",
+	}, envoye2e.ProxyE2ETests)
+	listener0 := params.LoadTestData("testdata/listener/client.yaml.tmpl")
+	params.Vars["StatsFilterClientConfig"] = driver.LoadTestJSON("testdata/stats/client_config_customized.yaml.tmpl")
+	listener1 := params.LoadTestData("testdata/listener/client.yaml.tmpl")
```
Comment (bianpengyuan) [inline test/envoye2e/stats_plugin/stats_test.go:380]: So a simple filter reload would cause this?

Comment (kyessenov) [reply test/envoye2e/stats_plugin/stats_test.go:380]: Yes. But it has to load without ANTLR first, then with it. The sequence is important.

**Comment by kyessenov** [comment]
Crash detected https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_proxy/2987/test_proxy/1075

**Pull request merged**